includes/embeds/class-amp-vimeo-embed-handler.php

```diff
@@ -137,42 +126,12 @@ public function render( $args ) {
 	 * @return int Video ID.
 	 */
 	private function get_video_id_from_url( $url ) {
-		$path = wp_parse_url( $url, PHP_URL_PATH );
-
 		// @todo This will not get the private key for unlisted videos (which look like https://vimeo.com/123456789/abcdef0123), but amp-vimeo doesn't support them currently anyway.
-		$video_id = '';
-		if ( $path && preg_match( ':/(\d+):', $path, $matches ) ) {
+		$video_id = null;
+		if ( preg_match( ':/video/(\d+):', $url, $matches ) ) {
```

Comment:
See note above about using constant and `strtok()`. It will be more robust than looking _anywhere_ in the URL for `/video/...` as it could erroneously match in a query parameter, for example.